stan/math/prim/prob/wiener4_lcdf.hpp

```diff
@@ -0,0 +1,863 @@
+#ifndef STAN_MATH_PRIM_PROB_WIENER4_LCDF_HPP
+#define STAN_MATH_PRIM_PROB_WIENER4_LCDF_HPP
+
+#include <stan/math/prim/prob/wiener5_lpdf.hpp>
+
+namespace stan {
+namespace math {
+namespace internal {
+
+/**
+ * Helper function. Calculate \f$\mathbb{e}^{x}\ if \f$x\ is not too large.
+ *
+ * @param x A scalar
+ * @return \f$\mathbb{e}^{x}\
+ */
+template <typename T_x>
+inline auto rexp(T_x&& x) noexcept {
+  return (x <= 700) ? exp(x) : exp(700);
+}
```

Comment:
Tagging @andrjohns do we do things like this anywhere else in Stan math? tmk we just check if it would overflow and if so we normally just return +/- inf or we let it overflow